File: TestNSData/TestNSData/ViewController.m
//
//  ViewController.m
//  TestNSData
//
//  Created by ICloud on 2020/5/25.
//  Copyright © 2020 ICloud. All rights reserved.
//
#import "InternalGlobalFunc.h"
#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    NSString *hexString = @"aa01010114";
    NSData *data = convertHexStrToData(hexString);
    
    Byte *testByte = [data bytes];

    for (int i = 0; i < data.length; i++) {
        NSLog(@"%x",testByte[i]);
        NSLog(@"%hhu",testByte[i]);
    }
    
    
    NSString *str = convertDataToHexStr(data);
    NSLog(@"%@", str);

}


@end

File: ShareSDKDemo/ShareSDKDemo/ViewController.swift
//
//  ViewController.swift
//  ShareSDKDemo
//
//  Created by ICloud on 2018/9/27.
//  Copyright © 2018年 ICloud. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func weChatLoginAction(_ sender: Any){
        if ShareSDK.hasAuthorized(.typeWechat){
            ShareSDK.cancelAuthorize(.typeWechat)
        }
        
        ShareSDK.getUserInfo(.typeWechat) { (state, user, error) in
            
        }
    }
    
    @IBAction func qqLoginAction(_ sender: Any){
        if ShareSDK.hasAuthorized(.typeQQ){
            ShareSDK.cancelAuthorize(.typeQQ)
        }
        
        ShareSDK.getUserInfo(.typeQQ) { (state, user, error) in
            
        }
    }
    
    @IBAction func weiboLoginAction(_ sender: Any){
        ShareSDK.getUserInfo(.typeSinaWeibo) { (state, user, error) in
            
        }
    }
    
    @IBAction func shareWechatAction(_ sender: Any) {
        
        let params = NSMutableDictionary()
        params.ssdkSetupShareParams(byText: "-----", images: UIImage(named: "lucky"), url: nil, title: nil, type: .image)
        ShareSDK.share(.subTypeWechatSession, parameters: params) { (state, data, content, error) in
            
        }
    }
    
    @IBAction func shareMomentsAction(_ sender: Any) {
        let params = NSMutableDictionary()
        params.ssdkSetupShareParams(byText: "-----", images: UIImage(named: "lucky"), url: nil, title: nil, type: .image)
        ShareSDK.share(.subTypeWechatTimeline, parameters: params) { (state, data, content, error) in
            
        }

    }
    @IBAction func shareWeiboAction(_ sender: Any) {
        
        let params = NSMutableDictionary()
        params.ssdkSetupShareParams(byText: "-----", images: UIImage(named: "lucky"), url: nil, title: nil, type: .image)
        ShareSDK.share(.typeSinaWeibo, parameters: params) { (state, data, content, error) in
            
        }
    }
    
    
    
}

